OpenSim/Common/DataTable.h

```diff
@@ -0,0 +1,1142 @@
+  /// indicates if there was a removal -- true means a key-value pair was
+  /// removed; false means the key was not found in metadata. Time complexity is
+  /// constant on average and linear in number of elements in the metadata on
+  /// worst case.
+  bool rmvMetaData(const string& key) {
+    return metadata.erase(key);
+  }
+
+  /// Clear the metadata. All the metadata will be lost with this operation.
+  void clearMetaData() {
+    metadata.clear();
+  }
+
+  /// Check if metadata for a given key exists. Time complexity is constant on
+  /// average and linear in the number of elements in the metadata on worst 
+  /// case.
+  bool metaDataExists(const string& key) const {
+    return metadata.find(key) != metadata.end();
+  }
+
+  /// Check if metadata is empty -- if the number of elements is zero.
+  bool isMetaDataEmpty() const {
+    return metadata.empty();
+  }
+
+  /// Get the number of elements in the metadata. Time complexity of other 
+  /// operations on metadata depend on this number.
+  size_t metaDataSize() const {
+    return metadata.size();
+  }
+
+  /*---------------------------------------------------------------------------*
+  * Column labels accessors & mutators.                                        *
+  *---------------------------------------------------------------------------*/
+
+  /// Check if a column index has label. Time complexity is linear in number of 
+  /// column labels. All columns will have an index. All columns need not have
+  /// a label.
+  ///
+  /// \throws OpenSim::ColumnDoesNotExist If col index specified does not exist.
+  bool colHasLabel(const size_t colind) const {
+    using ColLabelsValueType = typename ColLabelsType::value_type;
+    checkColExists(colind);
+    auto res = std::find_if(col_ind.begin(), 
+                            col_ind.end(), 
+                            [colind] (const ColLabelsValueType& kv) {
+                              return kv.second == colind;
+                            });
+    return res != col_ind.end();
+  }
+
+  /// Check if a column exists by its index.
+  bool colExists(const size_t colind) const {
+    return colind >= 0 && colind < static_cast<size_t>(data.ncol());
+  }
+
+  /// Check if a column exists under given label. All columns have an index but
+  /// not all columns may have labels.
+  bool colExists(const string& collabel) const {
+    return col_ind.find(collabel) != col_ind.end();
+  }
+
+  /// Label a column. The column should not have a label already. To update the
+  /// label of a column that already has a label, use updColLabel().
+  ///
+  /// \throws OpenSim::ColumnDoesNotExist If the col index specified does not
+  ///                                     exist.
+  /// \throws OpenSim::ColumnHasLabel If the column index specified already has
+  ///                                 a label.
+  void insertColLabel(const size_t colind, const string& collabel) {
```

Comment:
Sigh. OK, I have to admit this method name would look better as `insertColumnLabel()`. Maybe I just find it hard to read because of the two "L"s stuck together. Somehow I like col when dealing in data access, so that row and col are on an equal footing, but not as much when dealing with things that only apply to columns. Since I'm so wishy-washy about this I will leave it to others to sort out!